app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -103,8 +103,19 @@ export default class AMRHeatmapView extends React.Component {
     });
   }
 
-  correctGeneNames(filteredSamples) {
+  correctSampleAndGeneNames(filteredSamples) {
+    let sampleNames = {};
+    let duplicateSampleNames = new Set();
     filteredSamples.forEach(sample => {
+      // Keep track of samples with the same name, which may occur if
+      // a user selects samples from multiple projects.
+      if (Object.keys(sampleNames).includes(sample.sampleName)) {
+        sampleNames[sample.sampleName].push(sample);
+        duplicateSampleNames.add(sample.sampleName);
+      } else {
+        sampleNames[sample.sampleName] = [sample];
+      }
```

Comment:
You could use a JS [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of a separate Set and object - it allows you to store arbitrary keys while retaining the ability to iterate over it (with the `forEach` method) without having to dump them to an array.

Reply:
Switched to using a map instead of an object!

I kept the set since I want to use that to track only the duplicate samples so I can update only those names rather than iterating over the entire map of sample names again later.